### ISSUE TITLE
Don't allow primitive types as the initial signal input

### ISSUE THREAD

**Issue by saulshanabrook** [cerebral/action-tree#4]
In cerebral, we only want to allow signals to be called with objects, not primitive types (https://github.com/cerebral/cerebral/issues/252). If we wanted action tree to handle this for us, it should require an object as [the `initialPayload`](https://github.com/cerebral/action-tree/blob/534030e5b23c28842d1d7cb2e45c9a0397370974/src/executeTree.ts#L13) (and probably the subsequent `payload`). This is related to #2.

@Guria If this shouldn't be handled with `action-tree` can you reopen https://github.com/cerebral/cerebral/issues/252 ?

**Comment by Guria** [comment]
@christianalfoni what your opinion here? should we allow only serializable objects in `action-tree` or keep it unopinionated about it?

**Comment by Guria** [comment]
Oh. I think we can restrict to Object, no matter if it serializable or not in `action-tree` and keep checking in `cerbral` dev-mode that payload is serializable

**Comment by saulshanabrook** [comment]
I opened this one just to make sure the type is `Object`, not touching on serializability.

**Comment by Guria** [comment]
Yeah. We should check it here, because we can't merge primitives.
